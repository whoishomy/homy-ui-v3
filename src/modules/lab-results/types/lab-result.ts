import { LabResult as BaseLabResult } from '@/types/lab-results';

interface DataPoint {
  date: string;
  value: number;
}

export interface LabResult extends BaseLabResult {
  trendData?: Array<DataPoint>;
}
