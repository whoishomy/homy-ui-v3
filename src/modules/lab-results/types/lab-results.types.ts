export interface LabResult {
  id: string;
  patientId: string;
  testName: string;
  resultValue?: number;
  resultText?: string;
  referenceRange: string;
  unit: string;
  status: 'normal' | 'high' | 'low' | 'critical' | 'pending';
  testDate: string;
  reportedDate: string;
  doctorId?: string;
  insightComment?: string;
  aiConfidenceScore?: number;
}

export interface LabInsight {
  id: string;
  labResultId: string;
  insightType: 'trend' | 'anomaly' | 'recommendation';
  message: string;
  severity: 'info' | 'warning' | 'critical';
  confidence: number;
  generatedAt: string;
}

export interface PDFExportOptions {
  resultIds: string[];
  includeInsights: boolean;
  includeCharts: boolean;
  patientInfo: boolean;
  format: 'A4' | 'Letter';
  language: 'tr' | 'en';
}

export interface LabResultFilters {
  startDate?: string;
  endDate?: string;
  testTypes?: string[];
  status?: LabResult['status'][];
  searchTerm?: string;
}
