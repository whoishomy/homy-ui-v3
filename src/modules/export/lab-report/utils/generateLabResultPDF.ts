import { pdf, Document } from '@react-pdf/renderer';
import { createElement } from 'react';
import { LabResult } from '@/types/lab-results';
import { LabResultPDFTemplate } from '../components/LabResultPDFTemplate';
import { Filters } from '@/contexts/FilterContext';

interface GeneratePDFOptions {
  patientInfo?: {
    fullName: string;
    dateOfBirth: string;
    id: string;
  };
  fileName?: string;
  filters?: Filters;
}

export const generateLabResultPDF = async (
  result: LabResult,
  options: GeneratePDFOptions = {}
): Promise<Blob> => {
  const { patientInfo, fileName = `lab-result-${result.id}.pdf`, filters } = options;

  // Generate PDF blob
  const doc = createElement(
    Document,
    {},
    createElement(LabResultPDFTemplate, { result, patientInfo, filters })
  );
  const blob = await pdf(doc).toBlob();
  return blob;
};

export const downloadLabResultPDF = async (
  result: LabResult,
  options: GeneratePDFOptions = {}
): Promise<void> => {
  const { fileName = `lab-result-${result.id}.pdf` } = options;

  try {
    const blob = await generateLabResultPDF(result, options);
    const url = URL.createObjectURL(blob);
    const link = document.createElement('a');
    link.href = url;
    link.download = fileName;
    document.body.appendChild(link);
    link.click();
    document.body.removeChild(link);
    URL.revokeObjectURL(url);
  } catch (error) {
    console.error('PDF oluşturulurken bir hata oluştu:', error);
    throw error;
  }
};
