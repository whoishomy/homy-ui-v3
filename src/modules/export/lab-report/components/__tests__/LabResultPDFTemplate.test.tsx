import { render } from '@testing-library/react';
import { LabResultPDFTemplate } from '../LabResultPDFTemplate';
import { mockLabResult, mockFilters } from '@/test/utils';
import { AIGeneratedInsight } from '@/types/health-report';

// Mock react-pdf components
jest.mock('@react-pdf/renderer', () => ({
  Document: 'Document',
  Page: 'Page',
  Text: 'Text',
  View: 'View',
  StyleSheet: {
    create: (styles: any) => styles,
  },
  Font: {
    register: jest.fn(),
  },
  Image: 'Image',
}));

describe('LabResultPDFTemplate', () => {
  it('should render basic template with minimal props', () => {
    const { container } = render(<LabResultPDFTemplate result={mockLabResult} />);
    expect(container).toMatchSnapshot();
  });

  it('should render template with patient info', () => {
    const patientInfo = {
      fullName: 'John Doe',
      dateOfBirth: '1990-01-01',
      id: 'P12345',
    };

    const { container } = render(
      <LabResultPDFTemplate result={mockLabResult} patientInfo={patientInfo} />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render template with filters', () => {
    const { container } = render(
      <LabResultPDFTemplate result={mockLabResult} filters={mockFilters} />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render template with all props', () => {
    const patientInfo = {
      fullName: 'John Doe',
      dateOfBirth: '1990-01-01',
      id: 'P12345',
    };

    const { container } = render(
      <LabResultPDFTemplate
        result={mockLabResult}
        patientInfo={patientInfo}
        filters={mockFilters}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render insights when available', () => {
    const resultWithInsights = {
      ...mockLabResult,
      insights: [
        {
          type: 'summary' as AIGeneratedInsight['type'],
          content: 'Test insight',
          confidence: 0.95,
          sourceData: ['test'],
          timestamp: new Date().toISOString(),
        },
      ],
    };

    const { container } = render(<LabResultPDFTemplate result={resultWithInsights} />);
    expect(container).toMatchSnapshot();
  });

  it('should render notes when available', () => {
    const resultWithNotes = {
      ...mockLabResult,
      notes: 'Test notes',
    };

    const { container } = render(<LabResultPDFTemplate result={resultWithNotes} />);
    expect(container).toMatchSnapshot();
  });

  it('should render filter summary correctly', () => {
    const filters = {
      testType: 'Glucose',
      dateRange: '2024-03-20',
      trend: 'increasing' as const,
      significance: 'abnormal' as const,
    };

    const { container } = render(<LabResultPDFTemplate result={mockLabResult} filters={filters} />);
    expect(container).toMatchSnapshot();
  });
});
