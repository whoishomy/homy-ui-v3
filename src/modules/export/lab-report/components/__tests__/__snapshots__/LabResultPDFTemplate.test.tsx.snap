// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabResultPDFTemplate should render basic template with minimal props 1`] = `
<div>
  <document>
    <page
      style="flex-direction: column; background-color: rgb(255, 255, 255); padding: 30px; font-family: Inter;"
    >
      <view
        style="flex-direction: row; justify-content: space-between; margin-bottom: 20px; padding-bottom: 10px;"
      >
        <view
          style="flex-direction: column;"
        >
          <text
            style="font-size: 24px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 4px;"
          >
            Hemoglobin
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            20 Mart 2024
             • 
            Central Lab
          </text>
        </view>
        <view
          style="flex-direction: column; align-items: flex-end;"
        >
          <text>
            Rapor No: 
            test-uuid
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            İsteyen Hekim: 
            Dr. Mehmet Yılmaz
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          Test Sonuçları
        </text>
        <view
          style="flex-direction: row; align-items: center;"
        >
          <text
            style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
          >
            Parametre
          </text>
          <text
            style="flex: 1; font-size: 12px; font-weight: 500; color: rgb(17, 24, 39);"
          >
            Sonuç
          </text>
          <text
            style="flex: 1; font-size: 12px; color: rgb(107, 114, 128);"
          >
            Referans Aralığı
          </text>
          <text>
            Durum
          </text>
        </view>
        <view
          style="flex-direction: row; align-items: center;"
        >
          <text
            style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
          >
            Hemoglobin
          </text>
          <text
            style="flex: 1; font-size: 12px; font-weight: 500; color: rgb(17, 24, 39);"
          >
            14.5
             
            g/dL
          </text>
          <text
            style="flex: 1; font-size: 12px; color: rgb(107, 114, 128);"
          >
            12
             - 
            16
             
            g/dL
          </text>
          <text>
            Normal
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          AI İçgörüleri
        </text>
        <view
          style="padding: 8px; border-radius: 4px; background-color: rgb(243, 244, 246);"
        >
          <view
            style="flex-direction: row; align-items: center; margin-bottom: 4px;"
          >
            <text
              style="font-size: 10px; font-weight: 500; color: rgb(107, 114, 128); margin-right: 4px;"
            >
              Özet
            </text>
            <text>
              95
              % güven
            </text>
          </view>
          <text
            style="font-size: 11px; color: rgb(55, 65, 81);"
          >
            Hemoglobin değeri normal aralıkta ve stabil seyrediyor.
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          Notlar
        </text>
        <text
          style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
        >
          Normal referans aralığında
        </text>
      </view>
      <text
        style="position: absolute; bottom: 30px; left: 30px; right: 30px; font-size: 10px; color: rgb(107, 114, 128); text-align: center; padding-top: 10px;"
      >
        Bu rapor HomyOS Sağlık Sistemi tarafından 
        20 Mart 2024
         tarihinde oluşturulmuştur.
         • Onaylayan: Lab. Uzm. Ayşe Demir
      </text>
    </page>
  </document>
</div>
`;

exports[`LabResultPDFTemplate should render filter summary correctly 1`] = `
<div>
  <document>
    <page
      style="flex-direction: column; background-color: rgb(255, 255, 255); padding: 30px; font-family: Inter;"
    >
      <view
        style="flex-direction: row; justify-content: space-between; margin-bottom: 20px; padding-bottom: 10px;"
      >
        <view
          style="flex-direction: column;"
        >
          <text
            style="font-size: 24px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 4px;"
          >
            Hemoglobin
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            20 Mart 2024
             • 
            Central Lab
          </text>
          <text
            style="margin-top: 8px; font-size: 10px; color: rgb(107, 114, 128);"
          >
            Filtreler: Test: Glucose • Tarih: 2024-03-20 • Trend: increasing • Önem: abnormal
          </text>
        </view>
        <view
          style="flex-direction: column; align-items: flex-end;"
        >
          <text>
            Rapor No: 
            test-uuid
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            İsteyen Hekim: 
            Dr. Mehmet Yılmaz
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          Test Sonuçları
        </text>
        <view
          style="flex-direction: row; align-items: center;"
        >
          <text
            style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
          >
            Parametre
          </text>
          <text
            style="flex: 1; font-size: 12px; font-weight: 500; color: rgb(17, 24, 39);"
          >
            Sonuç
          </text>
          <text
            style="flex: 1; font-size: 12px; color: rgb(107, 114, 128);"
          >
            Referans Aralığı
          </text>
          <text>
            Durum
          </text>
        </view>
        <view
          style="flex-direction: row; align-items: center;"
        >
          <text
            style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
          >
            Hemoglobin
          </text>
          <text
            style="flex: 1; font-size: 12px; font-weight: 500; color: rgb(17, 24, 39);"
          >
            14.5
             
            g/dL
          </text>
          <text
            style="flex: 1; font-size: 12px; color: rgb(107, 114, 128);"
          >
            12
             - 
            16
             
            g/dL
          </text>
          <text>
            Normal
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          AI İçgörüleri
        </text>
        <view
          style="padding: 8px; border-radius: 4px; background-color: rgb(243, 244, 246);"
        >
          <view
            style="flex-direction: row; align-items: center; margin-bottom: 4px;"
          >
            <text
              style="font-size: 10px; font-weight: 500; color: rgb(107, 114, 128); margin-right: 4px;"
            >
              Özet
            </text>
            <text>
              95
              % güven
            </text>
          </view>
          <text
            style="font-size: 11px; color: rgb(55, 65, 81);"
          >
            Hemoglobin değeri normal aralıkta ve stabil seyrediyor.
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          Notlar
        </text>
        <text
          style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
        >
          Normal referans aralığında
        </text>
      </view>
      <text
        style="position: absolute; bottom: 30px; left: 30px; right: 30px; font-size: 10px; color: rgb(107, 114, 128); text-align: center; padding-top: 10px;"
      >
        Bu rapor HomyOS Sağlık Sistemi tarafından 
        20 Mart 2024
         tarihinde oluşturulmuştur.
         • Onaylayan: Lab. Uzm. Ayşe Demir
      </text>
    </page>
  </document>
</div>
`;

exports[`LabResultPDFTemplate should render insights when available 1`] = `
<div>
  <document>
    <page
      style="flex-direction: column; background-color: rgb(255, 255, 255); padding: 30px; font-family: Inter;"
    >
      <view
        style="flex-direction: row; justify-content: space-between; margin-bottom: 20px; padding-bottom: 10px;"
      >
        <view
          style="flex-direction: column;"
        >
          <text
            style="font-size: 24px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 4px;"
          >
            Hemoglobin
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            20 Mart 2024
             • 
            Central Lab
          </text>
        </view>
        <view
          style="flex-direction: column; align-items: flex-end;"
        >
          <text>
            Rapor No: 
            test-uuid
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            İsteyen Hekim: 
            Dr. Mehmet Yılmaz
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          Test Sonuçları
        </text>
        <view
          style="flex-direction: row; align-items: center;"
        >
          <text
            style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
          >
            Parametre
          </text>
          <text
            style="flex: 1; font-size: 12px; font-weight: 500; color: rgb(17, 24, 39);"
          >
            Sonuç
          </text>
          <text
            style="flex: 1; font-size: 12px; color: rgb(107, 114, 128);"
          >
            Referans Aralığı
          </text>
          <text>
            Durum
          </text>
        </view>
        <view
          style="flex-direction: row; align-items: center;"
        >
          <text
            style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
          >
            Hemoglobin
          </text>
          <text
            style="flex: 1; font-size: 12px; font-weight: 500; color: rgb(17, 24, 39);"
          >
            14.5
             
            g/dL
          </text>
          <text
            style="flex: 1; font-size: 12px; color: rgb(107, 114, 128);"
          >
            12
             - 
            16
             
            g/dL
          </text>
          <text>
            Normal
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          AI İçgörüleri
        </text>
        <view
          style="padding: 8px; border-radius: 4px; background-color: rgb(243, 244, 246);"
        >
          <view
            style="flex-direction: row; align-items: center; margin-bottom: 4px;"
          >
            <text
              style="font-size: 10px; font-weight: 500; color: rgb(107, 114, 128); margin-right: 4px;"
            >
              Özet
            </text>
            <text>
              95
              % güven
            </text>
          </view>
          <text
            style="font-size: 11px; color: rgb(55, 65, 81);"
          >
            Test insight
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          Notlar
        </text>
        <text
          style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
        >
          Normal referans aralığında
        </text>
      </view>
      <text
        style="position: absolute; bottom: 30px; left: 30px; right: 30px; font-size: 10px; color: rgb(107, 114, 128); text-align: center; padding-top: 10px;"
      >
        Bu rapor HomyOS Sağlık Sistemi tarafından 
        20 Mart 2024
         tarihinde oluşturulmuştur.
         • Onaylayan: Lab. Uzm. Ayşe Demir
      </text>
    </page>
  </document>
</div>
`;

exports[`LabResultPDFTemplate should render notes when available 1`] = `
<div>
  <document>
    <page
      style="flex-direction: column; background-color: rgb(255, 255, 255); padding: 30px; font-family: Inter;"
    >
      <view
        style="flex-direction: row; justify-content: space-between; margin-bottom: 20px; padding-bottom: 10px;"
      >
        <view
          style="flex-direction: column;"
        >
          <text
            style="font-size: 24px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 4px;"
          >
            Hemoglobin
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            20 Mart 2024
             • 
            Central Lab
          </text>
        </view>
        <view
          style="flex-direction: column; align-items: flex-end;"
        >
          <text>
            Rapor No: 
            test-uuid
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            İsteyen Hekim: 
            Dr. Mehmet Yılmaz
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          Test Sonuçları
        </text>
        <view
          style="flex-direction: row; align-items: center;"
        >
          <text
            style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
          >
            Parametre
          </text>
          <text
            style="flex: 1; font-size: 12px; font-weight: 500; color: rgb(17, 24, 39);"
          >
            Sonuç
          </text>
          <text
            style="flex: 1; font-size: 12px; color: rgb(107, 114, 128);"
          >
            Referans Aralığı
          </text>
          <text>
            Durum
          </text>
        </view>
        <view
          style="flex-direction: row; align-items: center;"
        >
          <text
            style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
          >
            Hemoglobin
          </text>
          <text
            style="flex: 1; font-size: 12px; font-weight: 500; color: rgb(17, 24, 39);"
          >
            14.5
             
            g/dL
          </text>
          <text
            style="flex: 1; font-size: 12px; color: rgb(107, 114, 128);"
          >
            12
             - 
            16
             
            g/dL
          </text>
          <text>
            Normal
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          AI İçgörüleri
        </text>
        <view
          style="padding: 8px; border-radius: 4px; background-color: rgb(243, 244, 246);"
        >
          <view
            style="flex-direction: row; align-items: center; margin-bottom: 4px;"
          >
            <text
              style="font-size: 10px; font-weight: 500; color: rgb(107, 114, 128); margin-right: 4px;"
            >
              Özet
            </text>
            <text>
              95
              % güven
            </text>
          </view>
          <text
            style="font-size: 11px; color: rgb(55, 65, 81);"
          >
            Hemoglobin değeri normal aralıkta ve stabil seyrediyor.
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          Notlar
        </text>
        <text
          style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
        >
          Test notes
        </text>
      </view>
      <text
        style="position: absolute; bottom: 30px; left: 30px; right: 30px; font-size: 10px; color: rgb(107, 114, 128); text-align: center; padding-top: 10px;"
      >
        Bu rapor HomyOS Sağlık Sistemi tarafından 
        20 Mart 2024
         tarihinde oluşturulmuştur.
         • Onaylayan: Lab. Uzm. Ayşe Demir
      </text>
    </page>
  </document>
</div>
`;

exports[`LabResultPDFTemplate should render template with all props 1`] = `
<div>
  <document>
    <page
      style="flex-direction: column; background-color: rgb(255, 255, 255); padding: 30px; font-family: Inter;"
    >
      <view
        style="flex-direction: row; justify-content: space-between; margin-bottom: 20px; padding-bottom: 10px;"
      >
        <view
          style="flex-direction: column;"
        >
          <text
            style="font-size: 24px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 4px;"
          >
            Hemoglobin
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            20 Mart 2024
             • 
            Central Lab
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            John Doe
             •
             
            1 Oca 1990
          </text>
          <text
            style="margin-top: 8px; font-size: 10px; color: rgb(107, 114, 128);"
          >
            Filtreler: Test: Hemoglobin • Tarih: 2024-03-20 • Trend: stable • Önem: normal
          </text>
        </view>
        <view
          style="flex-direction: column; align-items: flex-end;"
        >
          <text>
            Rapor No: 
            test-uuid
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            İsteyen Hekim: 
            Dr. Mehmet Yılmaz
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          Test Sonuçları
        </text>
        <view
          style="flex-direction: row; align-items: center;"
        >
          <text
            style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
          >
            Parametre
          </text>
          <text
            style="flex: 1; font-size: 12px; font-weight: 500; color: rgb(17, 24, 39);"
          >
            Sonuç
          </text>
          <text
            style="flex: 1; font-size: 12px; color: rgb(107, 114, 128);"
          >
            Referans Aralığı
          </text>
          <text>
            Durum
          </text>
        </view>
        <view
          style="flex-direction: row; align-items: center;"
        >
          <text
            style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
          >
            Hemoglobin
          </text>
          <text
            style="flex: 1; font-size: 12px; font-weight: 500; color: rgb(17, 24, 39);"
          >
            14.5
             
            g/dL
          </text>
          <text
            style="flex: 1; font-size: 12px; color: rgb(107, 114, 128);"
          >
            12
             - 
            16
             
            g/dL
          </text>
          <text>
            Normal
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          AI İçgörüleri
        </text>
        <view
          style="padding: 8px; border-radius: 4px; background-color: rgb(243, 244, 246);"
        >
          <view
            style="flex-direction: row; align-items: center; margin-bottom: 4px;"
          >
            <text
              style="font-size: 10px; font-weight: 500; color: rgb(107, 114, 128); margin-right: 4px;"
            >
              Özet
            </text>
            <text>
              95
              % güven
            </text>
          </view>
          <text
            style="font-size: 11px; color: rgb(55, 65, 81);"
          >
            Hemoglobin değeri normal aralıkta ve stabil seyrediyor.
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          Notlar
        </text>
        <text
          style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
        >
          Normal referans aralığında
        </text>
      </view>
      <text
        style="position: absolute; bottom: 30px; left: 30px; right: 30px; font-size: 10px; color: rgb(107, 114, 128); text-align: center; padding-top: 10px;"
      >
        Bu rapor HomyOS Sağlık Sistemi tarafından 
        20 Mart 2024
         tarihinde oluşturulmuştur.
         • Onaylayan: Lab. Uzm. Ayşe Demir
      </text>
    </page>
  </document>
</div>
`;

exports[`LabResultPDFTemplate should render template with filters 1`] = `
<div>
  <document>
    <page
      style="flex-direction: column; background-color: rgb(255, 255, 255); padding: 30px; font-family: Inter;"
    >
      <view
        style="flex-direction: row; justify-content: space-between; margin-bottom: 20px; padding-bottom: 10px;"
      >
        <view
          style="flex-direction: column;"
        >
          <text
            style="font-size: 24px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 4px;"
          >
            Hemoglobin
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            20 Mart 2024
             • 
            Central Lab
          </text>
          <text
            style="margin-top: 8px; font-size: 10px; color: rgb(107, 114, 128);"
          >
            Filtreler: Test: Hemoglobin • Tarih: 2024-03-20 • Trend: stable • Önem: normal
          </text>
        </view>
        <view
          style="flex-direction: column; align-items: flex-end;"
        >
          <text>
            Rapor No: 
            test-uuid
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            İsteyen Hekim: 
            Dr. Mehmet Yılmaz
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          Test Sonuçları
        </text>
        <view
          style="flex-direction: row; align-items: center;"
        >
          <text
            style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
          >
            Parametre
          </text>
          <text
            style="flex: 1; font-size: 12px; font-weight: 500; color: rgb(17, 24, 39);"
          >
            Sonuç
          </text>
          <text
            style="flex: 1; font-size: 12px; color: rgb(107, 114, 128);"
          >
            Referans Aralığı
          </text>
          <text>
            Durum
          </text>
        </view>
        <view
          style="flex-direction: row; align-items: center;"
        >
          <text
            style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
          >
            Hemoglobin
          </text>
          <text
            style="flex: 1; font-size: 12px; font-weight: 500; color: rgb(17, 24, 39);"
          >
            14.5
             
            g/dL
          </text>
          <text
            style="flex: 1; font-size: 12px; color: rgb(107, 114, 128);"
          >
            12
             - 
            16
             
            g/dL
          </text>
          <text>
            Normal
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          AI İçgörüleri
        </text>
        <view
          style="padding: 8px; border-radius: 4px; background-color: rgb(243, 244, 246);"
        >
          <view
            style="flex-direction: row; align-items: center; margin-bottom: 4px;"
          >
            <text
              style="font-size: 10px; font-weight: 500; color: rgb(107, 114, 128); margin-right: 4px;"
            >
              Özet
            </text>
            <text>
              95
              % güven
            </text>
          </view>
          <text
            style="font-size: 11px; color: rgb(55, 65, 81);"
          >
            Hemoglobin değeri normal aralıkta ve stabil seyrediyor.
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          Notlar
        </text>
        <text
          style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
        >
          Normal referans aralığında
        </text>
      </view>
      <text
        style="position: absolute; bottom: 30px; left: 30px; right: 30px; font-size: 10px; color: rgb(107, 114, 128); text-align: center; padding-top: 10px;"
      >
        Bu rapor HomyOS Sağlık Sistemi tarafından 
        20 Mart 2024
         tarihinde oluşturulmuştur.
         • Onaylayan: Lab. Uzm. Ayşe Demir
      </text>
    </page>
  </document>
</div>
`;

exports[`LabResultPDFTemplate should render template with patient info 1`] = `
<div>
  <document>
    <page
      style="flex-direction: column; background-color: rgb(255, 255, 255); padding: 30px; font-family: Inter;"
    >
      <view
        style="flex-direction: row; justify-content: space-between; margin-bottom: 20px; padding-bottom: 10px;"
      >
        <view
          style="flex-direction: column;"
        >
          <text
            style="font-size: 24px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 4px;"
          >
            Hemoglobin
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            20 Mart 2024
             • 
            Central Lab
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            John Doe
             •
             
            1 Oca 1990
          </text>
        </view>
        <view
          style="flex-direction: column; align-items: flex-end;"
        >
          <text>
            Rapor No: 
            test-uuid
          </text>
          <text
            style="font-size: 14px; color: rgb(107, 114, 128);"
          >
            İsteyen Hekim: 
            Dr. Mehmet Yılmaz
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          Test Sonuçları
        </text>
        <view
          style="flex-direction: row; align-items: center;"
        >
          <text
            style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
          >
            Parametre
          </text>
          <text
            style="flex: 1; font-size: 12px; font-weight: 500; color: rgb(17, 24, 39);"
          >
            Sonuç
          </text>
          <text
            style="flex: 1; font-size: 12px; color: rgb(107, 114, 128);"
          >
            Referans Aralığı
          </text>
          <text>
            Durum
          </text>
        </view>
        <view
          style="flex-direction: row; align-items: center;"
        >
          <text
            style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
          >
            Hemoglobin
          </text>
          <text
            style="flex: 1; font-size: 12px; font-weight: 500; color: rgb(17, 24, 39);"
          >
            14.5
             
            g/dL
          </text>
          <text
            style="flex: 1; font-size: 12px; color: rgb(107, 114, 128);"
          >
            12
             - 
            16
             
            g/dL
          </text>
          <text>
            Normal
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          AI İçgörüleri
        </text>
        <view
          style="padding: 8px; border-radius: 4px; background-color: rgb(243, 244, 246);"
        >
          <view
            style="flex-direction: row; align-items: center; margin-bottom: 4px;"
          >
            <text
              style="font-size: 10px; font-weight: 500; color: rgb(107, 114, 128); margin-right: 4px;"
            >
              Özet
            </text>
            <text>
              95
              % güven
            </text>
          </view>
          <text
            style="font-size: 11px; color: rgb(55, 65, 81);"
          >
            Hemoglobin değeri normal aralıkta ve stabil seyrediyor.
          </text>
        </view>
      </view>
      <view
        style="margin-top: 16px; margin-bottom: 16px;"
      >
        <text
          style="font-size: 16px; font-weight: 600; color: rgb(17, 24, 39); margin-bottom: 8px;"
        >
          Notlar
        </text>
        <text
          style="flex: 1; font-size: 12px; color: rgb(55, 65, 81);"
        >
          Normal referans aralığında
        </text>
      </view>
      <text
        style="position: absolute; bottom: 30px; left: 30px; right: 30px; font-size: 10px; color: rgb(107, 114, 128); text-align: center; padding-top: 10px;"
      >
        Bu rapor HomyOS Sağlık Sistemi tarafından 
        20 Mart 2024
         tarihinde oluşturulmuştur.
         • Onaylayan: Lab. Uzm. Ayşe Demir
      </text>
    </page>
  </document>
</div>
`;
